redbot/setup.py

```diff
@@ -253,6 +253,7 @@ def get_target_backend(backend) -> BackendType:
 
     backend = get_current_backend(instance)
     driver_cls = drivers.get_driver_class(backend)
+    await driver_cls.initialize(**data_manager.storage_details())
 
     if delete_data is True:
         await driver_cls.delete_all_data(interactive=interactive, drop_db=drop_db)
```

Comment:
Even though we drop data, *I think* we still need to run `teardown()` here to close pool.
```suggestion
         await driver_cls.delete_all_data(interactive=interactive, drop_db=drop_db)
     await driver_cls.teardown()
```